addons/crm/tests/test_crm_lead_duplicated.py

```diff
@@ -0,0 +1,182 @@
+            'name': 'Opportunity 4',
+            'type': 'opportunity',
+            'partner_name': 'Bob Doe'
+        })
+        leads[6] = self.env['crm.lead'].create({
+            'name': 'Opportunity 5',
+            'type': 'opportunity',
+            'partner_name': 'Bob Doe',
+            'email_from': 'bob@odoo.com',
+        })
+        leads[7] = self.env['crm.lead'].create({
+            'name': 'Opportunity 6',
+            'type': 'opportunity',
+            'email_from': 'bob@mymail.com',
+        })
+        leads[8] = self.env['crm.lead'].create({
+            'name': 'Opportunity 7',
+            'type': 'opportunity',
+            'email_from': 'alice@mymail.com',
+        })
+        leads[9] = self.env['crm.lead'].create({
+            'name': 'Opportunity 8',
+            'type': 'opportunity',
+            'probability': 0,
+            'active': False,
+            'email_from': 'alice@mymail.com',
+        })
+        leads[10] = self.env['crm.lead'].create({
+            'name': 'Opportunity 9',
+            'type': 'opportunity',
+            'contact_name': 'charlie'
+        })
+        leads[11] = self.env['crm.lead'].create({
+            'name': 'Opportunity 10',
+            'type': 'opportunity',
+            'contact_name': 'Charlie Chapelin',
+        })
+
+        self.company = self.env['res.partner'].create({
+            'name': 'My company',
+            'email': 'mycompany@company.com',
+            'is_company': True,
+            'street': '57th Street',
+            'city': 'New New York',
+            'country_id': self.env.ref('base.us').id,
+            'zip': '12345',
+        })
+
+        self.partner_1 = self.env['res.partner'].create({
+            'name': 'Dave',
+            'email': 'dave@odoo.com',
+            'mobile': '+1 202 555 0123',
+            'phone': False,
+            'parent_id': self.company.id,
+            'is_company': False,
+            'street': 'Pearl street',
+            'city': 'California',
+            'country_id': self.env.ref('base.us').id,
+            'zip': '95826',
+        })
+
+        self.partner_2 = self.env['res.partner'].create({
+            'name': 'Eve',
+            'email': 'eve@odoo.com',
+            'mobile': '+1 202 555 3210',
+            'phone': False,
+            'parent_id': self.company.id,
+            'is_company': False,
+            'street': 'Wall street',
+            'city': 'New York',
+            'country_id': self.env.ref('base.us').id,
+            'zip': '54321',
+        })
+
+        leads[12] = self.env['crm.lead'].create({
+            'name': 'Opportunity 8',
+            'type': 'opportunity',
+            'partner_id': self.partner_1.id
+        })
+        leads[13] = self.env['crm.lead'].create({
+            'name': 'Lead 3',
+            'type': 'lead',
+            'partner_id': self.partner_2.id
+        })
+
+        self.check_duplicates(leads, 0, [0, 1, 2])
```

Comment:
Linked to check_duplicates comment, this call is quite impossible to understand. You should have explicit records to give to your custom asserts so that we understand what you are testing.